### PR TITLE
java.util.UUID

### PR DESCRIPTION
**Pull request by neacao**
Completed implement functions inside UUID.

### CODE REVIEW

File: java/util/UUID/UUID.hpp [hunk 1]
```diff
@@ -38,124 +38,139 @@ namespace Java {
...
-             * @return
+             *
+             * @return int
              */
             int	variant();
 
             /**
              * The version number associated with this UUID.
+             *
              * @return
```
Comment (frangte) [inline java/util/UUID/UUID.hpp]: Missing  type variable return

File: java/util/UUID/UUID.cpp [hunk 2]
```diff
@@ -26,6 +26,34 @@
...
+}
+
+/**
+ * Private constructor which uses a byte array to construct the new UUID.
+ *
+ * @param Array<byte> data
+ * @return UUID
+ */
 UUID::UUID(Array<byte> data) {
     long msb = 0;
```
Comment (frangte) [inline java/util/UUID/UUID.cpp]: Please set variable name  easy to understand.

**Review by frangte — CHANGES_REQUESTED** [review]

**Review by loint — DISMISSED** [review]
PR should be rejected because of comments below

File: java/util/UUID/UUID.cpp [hunk 3]
```diff
@@ -26,64 +26,136 @@
...
+
+/**
+ * Returns val represented by the specified number of hex digits
+ *
+ * @param long value
+ * @param int digit
+ * @return String
+ */
+String UUID::digits(long value, int digit) {
+    long hi = 1L << (digit * 4);
```
Comment (dthongvl) [inline java/util/UUID/UUID.cpp]: Please rename hi to high

File: java/util/UUID/UUID.cpp [hunk 4]
```diff
@@ -26,64 +26,136 @@
...
+ * @param Array<byte> data
+ * @return UUID
+ */
 UUID::UUID(Array<byte> data) {
-    long msb = 0;
-    long lsb = 0;
+    long mostSigBits = 0;
+    long leastSigBits = 0;
 
     for (int i = 0; i < 8; ++i) {
```
Comment (dthongvl) [inline java/util/UUID/UUID.cpp]: Please create **int i** outside the loop and rename it to index or something make sense
and reuse it in next loop
https://github.com/foodtiny/native/blob/master/CONTRIBUTION.md

File: java/util/UUID/UUID.cpp [hunk 5]
```diff
@@ -26,64 +26,136 @@
...
+ * Default destructor
+ */
 UUID::~UUID() {
 }
 
+/**
+ * Compares this UUID with the specified UUID.
+ *
+ * @param UUID target
+ * @return int
```
Comment (dthongvl) [inline java/util/UUID/UUID.cpp:112]: Please add comment about the return value 1, -1, 0 or define const

File: java/util/UUID/UUID.cpp [hunk 6]
```diff
@@ -26,64 +26,136 @@
...
-    return this->timestamp;
-}
-
+/**
+ * Returns a hash code for this UUID.
+ *
+ * @return int
+ */
 int UUID::hashCode() {
     long hilo = this->mostSigBits ^ this->leastSigBits;
```
Comment (dthongvl) [inline java/util/UUID/UUID.cpp]: Please rename hilo T.T

File: java/util/UUID/UUID.cpp [hunk 7]
```diff
@@ -92,32 +164,93 @@ long UUID::node() {
...
+    if (components.length != 5) {
+        //FIXME: an exception
+        return UUID();
+    }
 
-// TODO: remove trap after fix toString/fromString
-String UUID::getTrap() {
-    return this->trap;
-};
+    for (int i=0; i<5; i++)
```
Comment (dthongvl) [inline java/util/UUID/UUID.cpp]: Create **int i** ouside loop

**Review by dthongvl — COMMENTED** [review]
Tu Boi testing

File: java/util/UUID/UUID.cpp [hunk 8]
```diff
@@ -92,32 +170,93 @@ long UUID::node() {
...
+/**
+ * Static factory to retrieve a type 4 (pseudo randomly generated) UUID.
+ *
+ * @return UUID
+ */
+UUID UUID::randomUUID() {
+    Array<byte> randomBytes = {0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0};
+
+    srand(time(0));
+    for (int i = 0; i < 16; ++i) {
```
Comment (dthongvl) [inline java/util/UUID/UUID.cpp]: Please change this

**Review by dthongvl — CHANGES_REQUESTED** [review]
Please implement public int clockSequence()

**Review by dthongvl — APPROVED** [review]
Approved

**Review by frangte — APPROVED** [review]

**Pull request merged**